File: Frontend/Admin/src/app/shared/models/vehicle.ts
export class Vehicle {
  id: number;
  vehicleBrand: string;
  vehicleModel: string;
  manufacturingDate: string;
  price: number;
  hasAirConditioning: boolean;
  pictureUrl: string;
  transmissionType: string;
  numberOfDoors: number;
  maxNumberOfPassengers: number;
  luggageCapacity: number;
  fuelType: string;
  baseDepositAmount: number;
  baseMaxDistance: number;
  reviewsScore: number;
  vehicleRegistrationNumber: string;
  isAvailable: boolean;
  extra100KmPrice: number;
  canLeaveCountry: boolean;

  constructor(vehicle: Vehicle) {
    this.id = vehicle.id;
    this.vehicleBrand = vehicle.vehicleBrand;
    this.vehicleModel = vehicle.vehicleModel;
    this.manufacturingDate = vehicle.manufacturingDate;
    this.price = vehicle.price;
    this.hasAirConditioning = vehicle.hasAirConditioning;
    this.pictureUrl = vehicle.pictureUrl;
    this.transmissionType = vehicle.transmissionType;
    this.numberOfDoors = vehicle.numberOfDoors;
    this.maxNumberOfPassengers = vehicle.maxNumberOfPassengers;
    this.luggageCapacity = vehicle.luggageCapacity;
    this.fuelType = vehicle.fuelType;
    this.baseDepositAmount = vehicle.baseDepositAmount;
    this.baseMaxDistance = vehicle.baseMaxDistance;
    this.reviewsScore = vehicle.reviewsScore;
    this.vehicleRegistrationNumber = vehicle.vehicleRegistrationNumber;
    this.isAvailable = vehicle.isAvailable;
    this.extra100KmPrice = vehicle.extra100KmPrice;
    this.canLeaveCountry = vehicle.canLeaveCountry;
  }
}

File: Backend/API/Shared/Entities/Vehicle.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Shared.Entities
{
    public class Vehicle : BaseEntity
    {
        public string OwnerId { get; set; }
        public VehicleModel VehicleModel { get; set; }

        public int VehicleModelId { get; set; }

        public List<PickupDropoffLocation> PickupDropoffLocations { get; set; }
        public List<CategoryWithVehicle> CategoryWithVehicles { get; set; }

        public DateTime ManufacturingDate { get; set; }

        public decimal Price { get; set; }

        public decimal BaseDepositAmount { get; set; }

        public decimal Extra100KmPrice { get; set; }

        public int BaseMaxDistance { get; set; }

        public decimal ReviewsScore { get; set; }

        public bool HasAirConditioning { get; set; }

        public string TransmissionType { get; set; }

        public int NumberOfDoors { get; set; }

        public int MaxNumberOfPassengers { get; set; }

        public int LuggageCapacity { get; set; }

        public string FuelType { get; set; }

        public string PictureUrl { get; set; }

        public string VehicleRegistrationNumber { get; set; }

        public bool IsAvailable { get; set; }

        public bool CanLeaveCountry { get; set; }
    }
}
